test_plans/oneapi_kernel_compiler_spirv.asciidoc

```diff
@@ -0,0 +1,45 @@
+:sectnums:
+:xrefstyle: short
+
+= Test plan for sycl_ext_oneapi_kernel_compiler_spirv
+
+This is a test plan for the API described in
+https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/experimental/sycl_ext_oneapi_kernel_compiler_spirv.asciidoc[sycl_ext_oneapi_kernel_compiler_spirv].
+
+
+== Testing scope
+
+=== Device coverage
+
+All of the tests described below are performed only on the default device that
+is selected on the CTS command line.
+
+=== Feature test macro
+
+All of the tests should use `#ifdef SYCL_EXT_ONEAPI_KERNEL_COMPILER_SPIRV` so they can be skipped
+if feature is not supported.
+
+== Tests
+
+All of the following tests run SPIR-V kernels loaded in binary form.
```

Comment:
This is probably more of a question to the extension spec:

It seems like it allows to obtain a (more or less) regular `kernel_bundle<executable>` that can be passed by user to `handler::use_kernel_bundle` API and define and launch a kernel using regular `parallel_for` mechanism in the SYCL source.

The question here: is it legal? If not, do we expect a certain exception to be thrown? And if so, there should be a test case for this.

Reply:
The extension spec says the following:

> The new APIs added by this extension are an expansion of the existing `kernel_bundle` capabilities. Thus, an application can create a kernel bundle from a source string and then build the bundle into "executable" bundle state. Once the application obtains a `kernel` object, it can use existing APIs from the core SYCL specification to set the value of kernel arguments and enqueue the kernel to a device.

(https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/experimental/sycl_ext_oneapi_kernel_compiler.asciidoc#overview)

Therefore, the application must use the overload of `parallel_for` or `single_task` that takes a parameter of type `kernel` if it wants to enqueue a kernel that was created from this extension.  These overloads are defined in the core SYCL specification, and the description contains the following text:

> This invocation function ignores any `kernel_bundle` that was bound to this command group handler via `handler::use_kernel_bundle()` and instead implicitly uses the kernel bundle that contains the `kernelObject`.

Therefore, it is not necessary to call `handler::use_kernel_bundle`.